WeakAuras/GenericTrigger.lua

```diff
@@ -3375,7 +3375,8 @@ function WeakAuras.GetSpellCost(powerTypeToCheck)
   if spellID then
     local costTable = GetSpellPowerCost(spellID);
     for _, costInfo in pairs(costTable) do
-      if costInfo.type == powerTypeToCheck then
+      -- When there is no required aura for a power cost, the API returns an aura ID of 0 and false for hasRequiredAura despite being valid.
+      if (costInfo.type == powerTypeToCheck) and (costInfo.requiredAuraID == 0) or costInfo.hasRequiredAura then
```

Comment:
The logic is wrong though, and binds stronger than, but we want type-match and ( == 0 or hasRequiredAura)

Reply:
Okay after reading a primer on operator precedence I think I've got it fixed and with no unnecessary parentheses.